### ISSUE TITLE
for/id associations for ThreePartDate

### ISSUE THREAD

**Issue by BryceStevenWilley** [SuffolkLITLab/docassemble-ALToolbox#214]
From VT's accessibility audit, relating to the ThreePartDateInput.

> On step 11, the “Month”, “Day”, and “Year” < label > elements are disassociated from their inputs
(seemingly deliberately, judging by the word “ignore” in their for attributes). As a result, screen reader
users cannot hear these labels. We recommend creating proper for / id associations. 3-Moderate

Here's an example chunk of the ThreePartDate HTML for reference.

```html
<div class="col-md-8 dafieldpart">
    <div class="al_three_parts_date form-row row">
        <div class="col col-month">
            <label for="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_month">Month</label>
            <select class="form-select al_field month ZGF0ZV9vZl9ncmFkdWF0aW9u" id="ZGF0ZV9vZl9ncmFkdWF0aW9u_month" name="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_month" aria-describedby="ZGF0ZV9vZl9ncmFkdWF0aW9u-error" aria-invalid="false">
                <option value=""></option>
                <option value="01">January</option>
                <option value="02">February</option>
                ...
            </select>
        </div>
        <div class="col col-3 col-day">
            <label for="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_day">Day</label>
            <input class="form-control al_field day ZGF0ZV9vZl9ncmFkdWF0aW9u" type="number" min="1" inputmode="numeric" id="ZGF0ZV9vZl9ncmFkdWF0aW9u_day" name="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_day" aria-describedby="ZGF0ZV9vZl9ncmFkdWF0aW9u-error">
        </div>
        <div class="col col-3 col-year">
            <label for="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_year">Year</label>
            <input class="form-control al_field year ZGF0ZV9vZl9ncmFkdWF0aW9u" type="number" min="1" inputmode="numeric" id="ZGF0ZV9vZl9ncmFkdWF0aW9u_year" name="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_year" aria-describedby="ZGF0ZV9vZl9ncmFkdWF0aW9u-error">
        </div>
        <div id="al_ZGF0ZV9vZl9ncmFkdWF0aW9u_error" class="al_error"></div>
    </div>
    <input value="02/3/2040" alt="Input box" class="form-control daThreePartsDate" data-almin="11/28/2023" data-almax="12/31/2040" type="hidden" name="ZGF0ZV9vZl9ncmFkdWF0aW9u" id="ZGF0ZV9vZl9ncmFkdWF0aW9u" style="display: none;" aria-hidden="true">
</div>
```

I'll have to look into what DA and HTML normally do with `name` compared to `id`, but I might need your help / perspective when building this @plocket

**Comment by BryceStevenWilley** [comment]
If you know what to do, go for it! I might do some extra testing before it goes in, but you are welcome to take this at your leisure, I won't be able to get back to it for a bit

**Comment by plocket** [comment]
Incremental progress: ALKiln should be fine with this change
